src/transform/CollapseTable.js

```diff
@@ -3,34 +3,102 @@ import Polyfill from './Polyfill'
 import elementUtilities from './ElementUtilities'
 
 const SECTION_TOGGLED_EVENT_TYPE = 'section-toggled'
+const ELEMENT_NODE = 1
+const TEXT_NODE = 3
+
+/**
+ * Determine if we want to extract text from this header.
+ * @param {!Element} header
+ * @return {!boolean}
+ */
+const isHeaderEligible = header => Polyfill.querySelectorAll(header, 'a').length < 3
+
+/**
+ * Determine eligibility of extracted text.
+ * @param {!string} headerText
+ * @param {?string} pageTitle
+ * @return {!boolean}
+ */
+const isHeaderTextEligible =
+  (headerText, pageTitle) => headerText && headerText !== pageTitle
+
+/**
+ * Determines whether a node has text which is non-comment, non-whitespace and non-zero length.
+ * Needed because some table headers have multiple text bearing child nodes and we need a way to
+ * find the first one. The following articles are examples:
+ *   'enwiki > Ramon Magsaysay High School, Manila'
+ *   'dewiki > Hornburg (Mansfelder Land)'
+ * @param  {!node} node
+ * @return {!boolean}
+ */
+const nodeHasNonWhitespaceTextContent = node => {
+  if (node.nodeType === TEXT_NODE || node.nodeType === ELEMENT_NODE) {
+    const text = node.textContent
+    if (!text) {
+      return false
+    }
+    return text.trim().length > 0
```

Comment:
A terser way to write these 5 lines is `return !!(node.textContent || '').length`.

Reply:
Doesn't catch whitespace only cases. 2 tests fail. I maybe tend a bit more toward the declarative end of the verbose/terse spectrum... have you been coding in Perl again ;)

Reply:
Ah, I missed the call to trim. Here you go... `return !!(node.textContent || '').trim().length`. Yeah, we have pretty different preferences when it comes to code (which is fine).

Reply:
The fix for the issues I was ironing out ended up removing this code.